libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -1007,6 +1020,68 @@ static void _serializeUnsubscribe( const IotMqttSubscription_t * pSubscriptionLi
 
 /*-----------------------------------------------------------*/
 
+static IotMqttError_t _decodeSubackStatus( size_t statusCount,
+                                           const uint8_t * pStatusStart,
+                                           _mqttPacket_t * pSuback )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    uint8_t subscriptionStatus = 0;
+    size_t i = 0;
+
+    /* Iterate through each status byte in the SUBACK packet. */
+    for( i = 0; i < statusCount; i++ )
+    {
+        /* Read a single status byte in SUBACK. */
+        subscriptionStatus = *( pStatusStart + i );
+
+        /* MQTT 3.1.1 defines the following values as status codes. */
+        switch( subscriptionStatus )
+        {
+            case 0x00:
```

Comment:
In general I'd prefer to see `enum` values here rather than integer values, but we won't hold this up for that since the code was already this way.